vm/src/builtins.rs

```diff
@@ -933,7 +935,20 @@ pub fn builtin_build_class_(
 
     let cells = vm.ctx.new_dict();
 
-    vm.invoke_with_locals(&function, cells.clone(), namespace.clone())?;
+    let PyFunction { code, scope, .. } = &*function;
+
+    let is_class = scope.is_class();
+
+    let mut scope = scope
+        .new_child_scope_with_locals(cells.clone())
+        .new_child_scope_with_locals(namespace.clone());
+
+    if is_class {
```

Comment:
I'm surprised by this ```if``` - aren't we always defining a class? (Or have I just not worked through this in enough detail.) I'd expect that we'd always mark the scope as a class here, and check it in function def. Do we need the compiler change?

Reply:
I did this because you can just call `builtins.__build_class__` with a function of your own, and then it doesn't change the scope:

```python
>>> a = 1
>>> def f():
...   a = 2
...   def b(): print(a)
...   b()
... 
>>> __build_class__(f, "A")
2
<class '__main__.A'>
```

Reply:
Ouch. Python!

In that case can the magic go in the LoadBuildClass instruction?